src/sentry/conf/server.py

```diff
@@ -2136,6 +2136,10 @@ def build_cdc_postgres_init_db_volume(settings):
         "custom_measurements": True,
     },
 }
+# How does production configure the Sentry DSN?
+# sentry/utils/sdk.py loads the upstream_dsn through this
+if os.environ.get("SENTRY_DSN"):
+    SENTRY_SDK_CONFIG["dsn"] = os.environ.get("SENTRY_DSN")
```

Comment:
i'm not sure how this works cc @markstory

Reply:
This is the code that picks up the DSN value:
https://github.com/getsentry/sentry/blob/0c1e21f3034dc43fea914729065bfbecb34a6feb/src/sentry/utils/sdk.py#L271-L276

It is then passed to the transport:
https://github.com/getsentry/sentry/blob/0c1e21f3034dc43fea914729065bfbecb34a6feb/src/sentry/utils/sdk.py#L283-L285

and eventually initialized:
https://github.com/getsentry/sentry/blob/0c1e21f3034dc43fea914729065bfbecb34a6feb/src/sentry/utils/sdk.py#L370-L382

Reply:
doesn't the sdk already read from SENTRY_DSN though https://github.com/getsentry/sentry-python/blob/96ea71f369f6e94241dc14647c21f1243e52cb6c/sentry_sdk/client.py#L65

Reply:
I've looked a bit more into this.
We don't use the env variable since we build our own transport.

In production, we add the `dsn` key to SENTRY_SDK_CONFIG, thus, it works in production.
https://github.com/getsentry/getsentry/blob/master/getsentry/conf/settings/prod.py#L595-L606

I will remove this block and think if using this env variable makes sense for development.